server/app/routes/results.py

```diff
@@ -32,54 +28,35 @@ def index(question_id):
                 user_id,
             )
             raise ForbiddenError()
-
-        stats = compute_stats(user_id, question["answer"])
-        if not stats:
-            raise BadRequestError()
-
-        question = write_stats(question_id, stats)
-
-        delete_local_file(f"{TMP_DIR}/{question_id}.TextGrid")
-        delete_local_file(f"{TMP_DIR}/{question_id}{WAV_EXT}")
-
         return {"result": question["stats"]}, 200
 
-    # @app.route("/api/results/<question_id>", methods=["GET"])
-    # def get_results(uid, type):
-    #     """
-    #     - files dir in = <uid>
-    #     - file names in user doc under bucket_files key
-    #     - fetch files, analyse audio
-    #     - aggregate stats
-    #     - return aggd and individual stats
-    #     """
-    #     user_id = get_user_cookie()
-    #     if not user_id:
-    #         raise ForbiddenError()
-
-    #     user_questions = get_questions_for_user(user_id)
-    #     if not user_questions:
-    #         raise NotFoundError()
-    # bucket_dir = uid
-    # all_stats = []
-
-    # for each in user["bucket_files"]:
-    #     file_path = f"{bucket_dir}/{each}"
-    #     transcript = get_transcript(f"gs://{GCS_BUCKET}/{file_path}")
-    #     stats = get_stats(transcript, each, TMP_DIR)
-    #     all_stats.append(stats)
+    @app.route("/api/questions/results", methods=["GET"])
+    def get_results_user():
+        user_id = get_user_cookie()
+        if not user_id:
+            raise ForbiddenError()
 
```

Comment:
I've seen this logic repeated across multiple endpoints. Its simple, but would we benefit by extracting this to some sort of exception handling file that checks common exceptions? We could pass the user_id each time, just to follow the DRY principle.

Also see lines 13-16 of this file, lines 9-11 of `task_status.py`. etc.

Reply:
Good catch. What do you think about the `ForbiddenError` exception being raised inside `get_user_cookie()? That way we can simply use `get_user_cookie()` for any existing future routes. If no cookie is found, the said exception is raised and handled by the error handler

Reply:
Exactly! Even better idea. That would be perfect.